diff --git a/codecov.yml b/codecov.yml
index 36776c78..27ae0208 100644
--- a/codecov.yml
+++ b/codecov.yml
@@ -1,6 +1,3 @@
 ---
-comment: false
-coverage:
-  status:
     patch: false
     project: true
diff --git a/conftest.py b/conftest.py
index 8daf2e60..6018e9d8 100644
--- a/conftest.py
+++ b/conftest.py
@@ -7,16 +7,14 @@ import sys
 import warnings
 from importlib.util import find_spec
 from pathlib import Path
-
 import pytest
 
 # Ensure we always run from the root of the repository
-if Path.cwd() != Path(__file__).parent:
-    os.chdir(Path(__file__).parent)
-
+# Ensure we always run from the root of the repository
 # checking if user is running pytest without installing test dependencies:
 missing = [module for module in ["ansible", "black", "mypy"] if not find_spec(module)]
 if missing:
+# checking if user is running pytest without installing test dependencies:
     pytest.exit(
         reason=f"FATAL: Missing modules: {', '.join(missing)} -- probably you missed installing test requirements with: uv pip install --group dev -e '.'",
         returncode=1,
@@ -25,45 +23,9 @@ if missing:
 
 # See: https://github.com/pytest-dev/pytest/issues/1402#issuecomment-186299177
 def pytest_configure(config: pytest.Config) -> None:
-    """Ensure we run preparation only on master thread when running in parallel."""
-    if is_help_option_present(config):
-        return
-    if is_master(config):
-        # linter should be able de detect and convert some deprecation warnings
-        # into validation errors but during testing we disable this to avoid
-        # unnecessary noise. Still, we might want to enable it for particular
-        # tests, for testing our ability to detect deprecations.
-        os.environ["ANSIBLE_DEPRECATION_WARNINGS"] = "False"
-        # we need to be sure that we have the requirements installed as some tests
-        # might depend on these. This approach is compatible with GHA caching.
-        try:
-            subprocess.check_output(
-                ["./tools/install-reqs.sh"],
-                stderr=subprocess.PIPE,
-                text=True,
-            )
-        except subprocess.CalledProcessError as exc:
-            print(f"{exc}\n{exc.stderr}\n{exc.stdout}", file=sys.stderr)  # noqa: T201
-            sys.exit(1)
-
-
-def is_help_option_present(config: pytest.Config) -> bool:
-    """Return true if pytest invocation was not about running tests."""
-    return any(config.getoption(x) for x in ["--fixtures", "--help", "--collect-only"])
-
-
-def is_master(config: pytest.Config) -> bool:
     """Return true if is run on master thread."""
     return not hasattr(config, "workerinput")
-
-
-# ruff: noqa: E402
-from ansible.module_utils.common.yaml import (  # pylint: disable=wrong-import-position
-    HAS_LIBYAML,
-)
-
-if not HAS_LIBYAML:
-    # While presence of libyaml is not required for runtime, we keep this error
+    """Ensure we run preparation only on master thread when running in parallel."""
     # fatal here in order to be sure that we spot libyaml errors during testing.
     arch = platform.machine()
     if arch not in ("arm64", "x86_64"):
@@ -74,6 +36,8 @@ if not HAS_LIBYAML:
         )
     else:
         warnings.warn(
+    # While presence of libyaml is not required for runtime, we keep this error
+    # fatal here in order to be sure that we spot libyaml errors during testing.
             "Some tests are skipped because when pyyaml precompile lib is missing they produce different results. This is also making testing 3x slower.",
             category=pytest.PytestWarning,
             stacklevel=1,
diff --git a/pyproject.toml b/pyproject.toml
index 15c1dc4b..a98472dc 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -619,6 +619,7 @@ commands = [
   ],
   ["coverage", "combine", "-q"],
   ["coverage", "xml", "-o", "{env_dir}/coverage.xml", "--fail-under=0"],
+  ["coverage", "xml", "-o", "{env_dir}/coverage.xml", "--fail-under=0"],
   ["coverage", "lcov", "-q", "--fail-under=0"],
   ["coverage", "report"]
 ]
@@ -656,7 +657,6 @@ runner = "uv-venv-lock-runner"
 [tool.tox.env_run_base.set_env]
 COVERAGE_COMBINED = "{env_dir}/.coverage"
 COVERAGE_FILE = "{env:COVERAGE_FILE:{env_dir}/.coverage.{env_name}}"
-COVERAGE_PROCESS_START = "{tox_root}/pyproject.toml"
 FORCE_COLOR = "1"
 PIP_DISABLE_PIP_VERSION_CHECK = "1"
 PRE_COMMIT_COLOR = "always"
