--- a/renovate.json
+++ b/renovate.json
@@ -6,6 +6,11 @@
       "allowedVersions": "<2.17",
       "description": "Platform compatibility constraint",
       "matchPackageNames": ["ansible-core"]
     },
+    {
+      "allowedVersions": "<38",
+      "description": "Platform compatibility constraint",
+      "matchPackageNames": ["cryptography"]
+    },
     {
       "allowedVersions": "<1.16",
--- a/.packit.yaml
+++ b/.packit.yaml
@@ -28,5 +28,11 @@
     targets:
       - fedora-rawhide-x86_64 # one on PR should be enough
 
+  - job: copr_build
+    trigger: commit
+    branch: gh-readonly-queue/.*
+    targets:
+      - fedora-rawhide-x86_64
+
--- a/mise.toml
+++ b/mise.toml
@@ -5,4 +5,4 @@
 UV_PROJECT_ENVIRONMENT = "{{ get_env(name='VIRTUAL_ENV', default='.venv') }}"
 
 [tools]
-python = "latest"
+python = "3.13"
